controllers/keda/scaledobject_controller.go

```diff
@@ -211,6 +210,7 @@ func (r *ScaledObjectReconciler) reconcileScaledObject(ctx context.Context, logg
 	pausedAnnotationFound := scaledObject.HasPausedAnnotation()
 	if pausedAnnotationFound {
 		scaledToPausedCount := true
+		metricscollector.RecordScalerPaused(scaledObject.Namespace, scaledObject.Name, scaledToPausedCount)
```

Comment:
here we should only record metrics when we change the status from runing -> paused once and then again once when we set from paused -> running

Reply:
set it in the code where we are (un)setting the condition

Reply:
this should go below following line `conditions.SetPausedCondition(metav1.ConditionTrue, kedav1alpha1.ScaledObjectConditionPausedReason, msg)`